caffe2/opt/fusion.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CAFFE2_OPT_FUSION_H_
+#define CAFFE2_OPT_FUSION_H_
+
+#include "caffe2/proto/caffe2.pb.h"
+
+namespace caffe2 {
+namespace opt {
+
+caffe2::NetDef fuseConvRelu(caffe2::NetDef net);
```

Comment:
I'd suggest we have `NNModule` in and `NNModule` out interface so that we can do multiple passes without converting between NNModel and NetDef back and forth.